### PR TITLE
Fix bug in polars parquet decoding handler

### PR DESCRIPTION
**Pull request by danpf**
# TL;DR
When trying to use `flytekitplugins-polars` I would get the error:
```
Message:
    Could not open Parquet input source '<Buffer>': Parquet file size is 0 bytes
User error.
```

Checking out the code, the `PolarsDataFrameToParquetEncodingHandler` actually places the parquet file at '{location}/00000', but it tries to read it at '{location}'

I think this is the easy solution, but maybe `PolarsDataFrameToParquetEncodingHandler` should be returning the '{location}/00000' instead of the parent directory? 

## Type
 - [x] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [ ] Code completed
 - [ ] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

### CODE REVIEW

**Review by pingsutw — COMMENTED** [review]
could you update this [test](https://github.com/flyteorg/flytekit/blob/da8d03118b8f5962a254512ac3a27bfd3f00680d/plugins/flytekit-polars/tests/test_polars_plugin_sd.py#L73)? To test the decoder, we should have a task that produces structured dataset, and pass it to the download stream task.

**Comment by danpf** [comment]
> could you update this [test](https://github.com/flyteorg/flytekit/blob/da8d03118b8f5962a254512ac3a27bfd3f00680d/plugins/flytekit-polars/tests/test_polars_plugin_sd.py#L73)? To test the decoder, we should have a task that produces structured dataset, and pass it to the download stream task.

@pingsutw I'm not exactly sure what you mean by this.  But I added a very basic roundtrip style test, and I merged with master.

After looking at the code more closely, I believe that this current implementation is more in line with what was intended.  I also suspect that using pandas in the tests was unintentional (but i could be wrong), so I changed that (but it works either way)

Also, for the html representation of the data, I wonder if pl.DataFrame._repr_html_ might be a better fit?

File: plugins/flytekit-polars/flytekitplugins/polars/sd_transformers.py [hunk 1]
```diff
@@ -40,23 +40,32 @@ def encode(
         structured_dataset_type: StructuredDatasetType,
     ) -> literals.StructuredDataset:
         df = typing.cast(pl.DataFrame, structured_dataset.dataframe)
-
+        random_string = ctx.file_access.get_random_string()
```
Comment (kumare3) [inline plugins/flytekit-polars/flytekitplugins/polars/sd_transformers.py:43]: why do we need the random string here?

File: plugins/flytekit-polars/flytekitplugins/polars/sd_transformers.py [hunk 2]
```diff
@@ -40,23 +40,32 @@ def encode(
...
+                remote_dir,
+                random_string,
+            )
+        else:
+            remote_dir = ctx.file_access.get_random_remote_directory()
+            sd_uri = ctx.file_access.join(
+                remote_dir,
+                random_string,
             )
         ctx.file_access.upload_directory(local_dir, remote_dir)
```
Comment (kumare3) [inline plugins/flytekit-polars/flytekitplugins/polars/sd_transformers.py:67]: i actually want to get rid of this dual writes, one to local directory and then to remote. can we simply write to remote as a stream?

File: plugins/flytekit-polars/flytekitplugins/polars/sd_transformers.py (hunk 1 above)
Comment (danpf) [reply plugins/flytekit-polars/flytekitplugins/polars/sd_transformers.py]: I changed this back to the default `00000` parquet filename

File: plugins/flytekit-polars/flytekitplugins/polars/sd_transformers.py (hunk 2 above)
Comment (danpf) [reply plugins/flytekit-polars/flytekitplugins/polars/sd_transformers.py]: I've updated the code to write to a BytesIO stream. Let me know if that is in line with what you were thinking!

**Review by pingsutw — APPROVED** [review]

**Comment by pingsutw** [comment]
thank you @danpf
